ts/Stock/Indicators/CMO/CMOIndicator.ts

```diff
@@ -0,0 +1,198 @@
+/* *
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import type IndicatorValuesObject from '../IndicatorValuesObject';
+import type LineSeries from '../../../Series/Line/LineSeries';
+import type {
+    CMOOptions,
+    CMOParamsOptions
+} from './CMOOptions';
+import type CMOPoint from './CMOPoint';
+
+import SeriesRegistry from '../../../Core/Series/SeriesRegistry.js';
+const {
+    seriesTypes: {
+        sma: SMAIndicator
+    }
+} = SeriesRegistry;
+import U from '../../../Core/Utilities.js';
+const {
+    isNumber,
+    merge
+} = U;
+
+/* eslint-disable require-jsdoc */
+
+// Utils:
+function toFixed(a: number, n: number): number {
+    return parseFloat(a.toFixed(n));
+}
+/* eslint-enable require-jsdoc */
+
+/**
+ * The CMO series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.cmo
+ *
+ * @augments Highcharts.Series
+ */
+class CMOIndicator extends SMAIndicator {
+    /**
+     * Chande Momentum Oscilator (CMO) technical indicator. This series
+     * requires the `linkedTo` option to be set and should be loaded after
+     * the `stock/indicators/indicators.js` file.
+     *
+     * @sample stock/indicators/cmo
+     *         CMO indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        next
+     * @product      highstock
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/cmo
+     * @optionparent plotOptions.cmo
+     */
+    public static defaultOptions: CMOOptions = merge(SMAIndicator.defaultOptions, {
+        params: {
+            period: 20,
+            index: 3
+        }
+    } as CMOOptions);
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public data: Array<CMOPoint> = void 0 as any;
+    public points: Array<CMOPoint> = void 0 as any;
+    public options: CMOOptions = void 0 as any;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public calculateSums(
+        period: number,
+        i: number,
+        values: Array<number>
+    ): { sumOfLowerValues: number; sumOfHigherValues: number } {
+        let sumOfLowerValues = 0,
+            sumOfHigherValues = 0;
+
+
+        for (let j = i; j > i - period; j--) {
+            if (values[j] > values[j - 1]) {
+                sumOfHigherValues += values[j] - values[j - 1];
+            } else if (values[j] < values[j - 1]) {
+                sumOfLowerValues += values[j - 1] - values[j];
+            }
+        }
+
+        return { sumOfLowerValues, sumOfHigherValues };
+    }
+
+    public getValues<TLinkedSeries extends LineSeries>(
+        series: TLinkedSeries,
+        params: CMOParamsOptions
+    ): IndicatorValuesObject<TLinkedSeries> | undefined {
+        var period = params.period as any,
```

Comment:
We might split them with `const/let` to better see which can be modified and which not.